commit SQL

goldovskaya-m opened the pull request.

goldovskaya-m committed:
commit SQL

AleksandrZhukovJava commented on src/main/java/sky/pro/Hogwarts31Test/repository/AvatarRepository.java:
```diff
@@ -8,4 +10,6 @@
 public interface AvatarRepository extends JpaRepository<Avatar, Long> {
     Optional<Avatar> findByStudentId(long studentId);
 
+    Page<Avatar> findAll(Pageable pageable);
```
Не нужно явно определять, его можно использовать из пакета

AleksandrZhukovJava commented on src/main/java/sky/pro/Hogwarts31Test/service/StudentService.java:
```diff
@@ -19,4 +20,10 @@ public interface StudentService {
+
+    Double getAverageAgeStudent();
+
+    Page<Student> fiveLastStudents();
```
findFiveLastStudents()

AleksandrZhukovJava commented on src/main/java/sky/pro/Hogwarts31Test/repository/StudentRepository.java:
```diff
@@ -10,4 +14,14 @@ public interface StudentRepository extends JpaRepository<Student, Long> {
+    Double getAverageAgeStudent();
+
+    @Query("SELECT s FROM Student s ORDER BY s.id DESC")
+    Page<Student> findLastFiveStudents(Pageable pageable);
```
Тут не нужен Page, тут нужен List и не принимать параметр

AleksandrZhukovJava commented on src/main/java/sky/pro/Hogwarts31Test/repository/StudentRepository.java:
```diff
@@ -10,4 +14,14 @@ public interface StudentRepository extends JpaRepository<Student, Long> {
+    @Query(value = "SELECT AVG(age) FROM Student", nativeQuery = true)
+    Double getAverageAgeStudent();
+
+    @Query("SELECT s FROM Student s ORDER BY s.id DESC")
```
не хватает limit 5

AleksandrZhukovJava commented on src/main/java/sky/pro/Hogwarts31Test/service/StudentServiceImpl.java:
```diff
@@ -80,4 +82,21 @@ public Long checkStudentExist(Long id) {
+    @Override
+    @Transactional
+    public Page<Student> fiveLastStudents() {
+        Pageable pageable = PageRequest.of(0, 5);
+        return studentRepository.findLastFiveStudents(pageable);
```
Это будет работать, но зачем эти сложности если запрос может по умолчанию возвращать 5 студентов

goldovskaya-m committed:
commit
Создан эндпоинт GET /students/print-parallel.
Эндпоинт GET /students/print-parallel распечатывает имена в консоль.
Печать в консоль происходит в многопоточном режиме.
Создан эндпоинт GET /students/print-synchronized.
Эндпоинт GET /students/print-synchronized распечатывает имена в консоль.
Создан синхронизированный метод для вывода имен студентов в консоль.
Печать в консоль происходит с помощью синхронизрованного метода.
Вызов синхронизированного метода просиходит из нескольких потоков.
